app/controllers/api/submission_files_controller.rb

```diff
@@ -168,5 +170,37 @@ def remove_file
         render 'shared/http_status', locals: { code: '500', message: message }, status: 500
       end
     end
+
+    def remove_folder
+      grouping = Grouping.find_by_group_id_and_assignment_id(params[:group_id], params[:assignment_id])
+      if grouping.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+            'No group with that id exists for the given assignment' }, status: 404
+        return
+      end
+
+      if has_missing_params?([:folder_path])
+        # incomplete/invalid HTTP params
+        render 'shared/http_status', locals: { code: '422', message:
+            HttpStatusHelper::ERROR_CODE['message']['422'] }, status: 422
+        return
+      end
+      success, messages = grouping.group.access_repo do |repo|
+        folders = params[:folder_path]
```

Comment:
we probably want to use the singular `folder` here since there will only be one of them